cloudinstall/charms/__init__.py

```diff
@@ -104,6 +104,9 @@ class CharmBase:
     menuable = False
     subordinate = False
     openstack_release_min = 'i'
+    # Set this if a plugin charm overrides a 'system' installed
+    # charm
+    tainted = False
```

Comment:
Is this used anywhere? If it's just for debugging, it might not be necessary, you could always look at the class's __file__ member.